tracing-subscriber/src/fmt/mod.rs

```diff
@@ -133,53 +137,46 @@ pub struct Subscriber<
 
 /// A `Subscriber` that logs formatted representations of `tracing` events.
 /// This type only logs formatted events; it does not perform any filtering.
-#[derive(Debug)]
-pub struct Formatter<
+pub type Formatter<
     N = format::DefaultFields,
     E = format::Format<format::Full>,
     W = fn() -> io::Stdout,
-> {
-    fmt_fields: N,
-    fmt_event: E,
-    spans: span::Store,
-    settings: Settings,
-    make_writer: W,
-}
+> = layer::Layered<FmtLayer<Registry, N, E, W>, Registry>;
 
 /// Configures and constructs `Subscriber`s.
 #[derive(Debug)]
-pub struct Builder<
+pub struct SubscriberBuilder<
     N = format::DefaultFields,
     E = format::Format<format::Full>,
     F = LevelFilter,
     W = fn() -> io::Stdout,
 > {
     filter: F,
-    fmt_fields: N,
-    fmt_event: E,
-    settings: Settings,
-    make_writer: W,
+    inner: LayerBuilder<Registry, N, E, W>,
 }
 
-#[derive(Debug)]
-struct Settings {
-    inherit_fields: bool,
-    initial_span_capacity: usize,
-}
+/// Configures and constructs `Subscriber`s.
+#[deprecated(since = "0.2.0", note = "renamed to `SubscriberBuilder`")]
+pub type Builder<
```

Comment:
Why this rename?

Reply:
It's @hawkw's code; just moved around as part of this PR.

Reply:
the rename is so that the subscriber `Builder` and layer `Builder` are easier to distinguish, I thought it was better to have a `LayerBuilder` and a `SubscriberBuilder` , rather than a `LayerBuilder` and a `Builder`. what do you think?

Reply:
My personal preference is to call them _both_ `Builder` and have them in different modules.

Reply:
I wanted to have a flatter module structure, so I thought we should make the layer module private and re-export. I could be convinced otherwise!